apps/explorer/src/hooks/useSearch.ts

```diff
@@ -108,9 +119,42 @@ const getResultsForAddress = async (rpc: JsonRpcProvider, query: string) => {
     };
 };
 
+// Using address to query by pool id or sui address
+const getResultsForValidatorByPoolIdOrSuiAddress = async (
+    rpc: JsonRpcProvider,
+    query: string
+) => {
+    const normalized = normalizeSuiObjectId(query);
+    if (!isValidSuiObjectId(normalized)) return null;
+
+    // Rather than calling the rpc method directly, we can use the queryClient to cache system state.
+    const { activeValidators } = await queryClient.fetchQuery(
```

Comment:
Should we use `useGetSystemState` from core instead since we are calling the same `getLatestSuiSystemState` method? We might need to pass `activeValidators` back into this function though.

Reply:
good point.